pkg/client/unversioned/clientcmd/merged_client_builder.go

```diff
@@ -112,8 +112,11 @@ func (config *DeferredLoadingClientConfig) ClientConfig() (*restclient.Config, e
 		//   "empty due to defaults"
 		// TODO: this shouldn't be a global - the client config rules should be
 		//   handling this.
-		defaultConfig, err := DefaultClientConfig.ClientConfig()
-		if err == nil && !reflect.DeepEqual(mergedConfig, defaultConfig) {
+		defaultConfig, defErr := DefaultClientConfig.ClientConfig()
+		if IsConfigurationInvalid(defErr) && !IsEmptyConfig(err) {
```

Comment:
> The reason I am being extra picky here is that this has already broken twice because it's hard to reason about what can happen.

@smarterclayton that's fair and makes sense.

> I'd like to not have this be the third time - if we fix federation and break kubectl, that's not much better.

That would be a much worse situation indeed :smile: 

Thanks for the LGTM!